### PR TITLE
SECRES-1649 - Address a libinjection false positive

### PR DESCRIPTION
**Pull request by Taiki-San**
libinjection interpret invalid sequences (such as words starting after a number without separators, which is causing false positives.

### CODE REVIEW

File: tests/matcher/is_sqli_test.cpp [hunk 1]
```diff
@@ -31,12 +31,22 @@ TEST(TestIsSQLi, TestNoMatch)
...
+    };
+    
+    for (auto pattern : no_match) {
+        std::cout << pattern << std::endl;
+        ddwaf_object param;
+        ddwaf_object_string(&param, pattern);
+
+        auto match = matcher.match(param);
+        EXPECT_FALSE(match.first);
+        std::cout << match.second << std::endl;
```
Comment (Anilm3) [inline tests/matcher/is_sqli_test.cpp:46]: Please remove the print statements.

**Review by Anilm3 — CHANGES_REQUESTED** [review]
You'll need to format it using clang-17:

```
cmake .. -DCLANG_FORMAT=/usr/bin/clang-format-17
make format_fix
```

I can do it for you if you need.

File: tests/matcher/is_sqli_test.cpp (hunk 1 above)
Comment (Taiki-San) [reply tests/matcher/is_sqli_test.cpp:46]: Oops, sorry!

File: src/vendor/libinjection/src/libinjection_sqli.c [hunk 2]
```diff
@@ -314,6 +302,12 @@ static void st_assign(stoken_t * st, const char stype,
     st->val[last] = CHAR_NULL;
 }
 
+static void sf_update_state(struct libinjection_sqli_state * sf, const char stype,
+                            size_t pos, size_t len, const char* value) {
+    st_assign(sf->current, stype, pos, len, value);
+    sf->hadWordBoundary = stype == TYPE_COMMENT;
```
Comment (Anilm3) [inline src/vendor/libinjection/src/libinjection_sqli.c:308]: `st_assign` uses `memcpy` while `st_assign_char` only had assignments. I think that's why the microbenchmark increased quite a bit:

```
scenario:is_sqli_matcher.random
🟥 execution_time [+477.773µs; +478.102µs] or [+643.703%; +644.145%]
```

**Review by Anilm3 — APPROVED** [review]

**Pull request merged**